compiler/stable_mir/src/mir/alloc.rs

```diff
@@ -16,7 +16,9 @@ pub enum GlobalAlloc {
     /// The alloc ID is used as a function pointer.
     Function(Instance),
     /// This alloc ID points to a symbolic (not-reified) vtable.
-    /// The `None` trait ref is used to represent auto traits.
+    /// We remember the full dyn type, not just the principal trait, so that
+    /// const-eval and Miri can detect UB due to invalid transmutes of
+    /// `dyn Trait` types.
```

Comment:
The comment does not seem to reflect reality?

Reply:
Oh, yeah, copy paste error lmao

Reply:
r=me with that fixed and the other two nits